src/Fmt.ml

```diff
@@ -150,15 +156,42 @@ let wrap_fits_breaks ?(space = true) conf x =
 
 (** Boxes ---------------------------------------------------------------*)
 
-let open_box n fs = Format.pp_open_box fs n
+let box_debug_enabled = ref false
+
+let box_stack = ref []
+
+let box_depth_colors = [|32; 33; 34; 31; 35; 36|]
+
+let box_depth_color () =
+  let depth = List.length !box_stack in
+  box_depth_colors.(depth % Array.length box_depth_colors)
+
+let debug_box_open open_sym close_sym fs =
+  if !box_debug_enabled then (
+    pp_color (box_depth_color ()) (fun fs -> fmt "@<0>%s" fs open_sym) fs ;
```

Comment:
If you end up passing the format stream around explicitly, it is probably clearer to not use the `fmt` combinators and just use `Format.fprintf` directly.

Reply:
`Format.fprintf` takes the `fs` as first parameter so it would not be much better.

Reply:
I agree.